File: 0128-longest-consecutive-sequence/0128-longest-consecutive-sequence.cpp
class Solution {
public:
    int longestConsecutive(vector<int>& nums) {
        set<int> st(nums.begin(),nums.end());
        if(nums.size()==0)return 0;
        int prev=nums[0];
        int m=0;
        int s=0;
        for(int i:st)
        {
            if(i!=prev+1)
            {
                m=max(m,s);
                s=0;
            }
            s++;
            prev=i;
        }
        m=max(m,s);
        return m;
    }
};

File: 0128-longest-consecutive-sequence/0128-longest-consecutive-sequence.java
class Solution {
    public int longestConsecutive(int[] nums) {
        int n=nums.length;
        HashSet<Integer> hs=new HashSet<>();
        for(int i=0;i<n;i++)
        {
            hs.add(nums[i]);
        }
        int maxi=0;
        for(int s:hs)
        {
            if(!hs.contains(s+1))
            {int cnt=1;
            while(hs.contains(--s))
            {
                cnt++;
            }
            maxi=Math.max(maxi,cnt);
            }
        }
        return maxi;
    }
}